code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -56,8 +58,34 @@
 						return
 					M.put_in_l_hand(src)
 			add_fingerprint(usr)
-			return
 
+//Case by case proc for Contents transfer to other containers, bins, turf, and more possible additions.
+/obj/item/weapon/storage/proc/contentto(obj/dest_object, mob/user)
+	playsound(loc, "rustle", 50, 1, -5)
+	if( istype(dest_object, /obj/item/weapon/storage) ) //Check if storage item
+		var/obj/item/weapon/storage/S = dest_object
+		for(var/obj/item/I in src)
+			if(S.can_be_inserted(I,0,user))
+				remove_from_storage(I, S)
+		orient2hud(user)
+		S.orient2hud(user)
+		if(user.s_active) //refresh the HUD to show the transfered contents
+			user.s_active.close(user)
+			user.s_active.show_to(user)
+		return
+
+	if( istype(dest_object, /obj/machinery/disposal) ) //Check if it's a bin
+		for(var/obj/item/I in src)
+			remove_from_storage(I, dest_object) //No check needed, put everything inside
+		return
+
+	var/turf/T = get_turf(dest_object) //Get_turf, and check if the turf isn't a wall
+	if( istype(T, /turf/simulated/wall) )
+		return
+
+	for(var/obj/item/I in src)
+		remove_from_storage(I, T)
+	return

```

Comment:
No need to return, you're not returning any value, however you could have ```storage_contents_dump_act()``` return 1 on success and 0 on failure, and then have this proc simply return that value aswell.